src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -138,27 +126,33 @@ private ArmSubsystem() {
     updatePID1Settings();
 
     burnFlash();
-    configureSpark("Arm set CANTimeout", () -> m_arm.setCANTimeout(0));
+    m_arm.setCANTimeout(0);
 
     ErrorTrackingSubsystem.getInstance().register(m_arm);
   }
 
   public void updatePID0Settings() {
-    configureSpark("Arm set FF", () -> (m_pidControllerArm.setFF(m_armFFSubs.get(), 0)));
-    configureSpark("Arm set P", () -> (m_pidControllerArm.setP(m_armPSubs.get(), 0)));
-    configureSpark("Arm set I", () -> (m_pidControllerArm.setI(m_armISubs.get(), 0)));
-    configureSpark("Arm set D", () -> (m_pidControllerArm.setD(m_armDSubs.get(), 0)));
-    configureSpark("Arm set Iz", () -> (m_pidControllerArm.setIZone(m_armIzSubs.get(), 0)));
-    configureSpark("Arm set Output Range",
-        () -> (m_pidControllerArm.setOutputRange(Config.ArmConfig.min_output, Config.ArmConfig.max_output)));
+    SparkMaxConfig m_arm_config = new SparkMaxConfig();
+    m_arm_config.closedLoop.velocityFF(m_armFFSubs.get(), ClosedLoopSlot.kSlot0);
+    m_arm_config.closedLoop.p(m_armPSubs.get(), ClosedLoopSlot.kSlot0);
+    m_arm_config.closedLoop.i(m_armPSubs.get(), ClosedLoopSlot.kSlot0);
+    m_arm_config.closedLoop.d(m_armDSubs.get(), ClosedLoopSlot.kSlot0);
+    m_arm_config.closedLoop.iZone(m_armIzSubs.get(), ClosedLoopSlot.kSlot0);
+    m_arm_config.closedLoop.outputRange(Config.ArmConfig.min_output, Config.ArmConfig.max_output);
+
+    m_arm.configure(m_arm_config, SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
```

Comment:
should have a common m_arm_config, which is updated by multiple functions and only call configure() once, instead of a new one.